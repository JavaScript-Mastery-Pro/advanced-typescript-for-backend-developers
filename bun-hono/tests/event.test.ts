import app from "..";

describe("Event", () => {
  test("GET /api/v1/events", async () => {
    const res = await app.request("/api/v1/events");
    expect(res.status).toBe(200);
    expect(await res.json()).toMatchSnapshot();
  });
});
