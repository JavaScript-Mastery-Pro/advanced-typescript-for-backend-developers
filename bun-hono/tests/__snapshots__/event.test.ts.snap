// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Example GET /api/v1/events 1`] = `
{
  "data": [
    {
      "date": "2024-12-01",
      "description": "A discussion about the future of AI and its impact on society.",
      "event_type": "talk",
      "id": 6,
      "title": "Tech Talk: AI in 2024",
      "userId": 3,
    },
    {
      "date": "2024-12-05",
      "description": "Hands-on coding session for developers new to JavaScript.",
      "event_type": "workshop",
      "id": 7,
      "title": "JavaScript Workshop",
      "userId": 2,
    },
    {
      "date": "2024-12-10",
      "description": "A beginner's guide to building web apps with React.",
      "event_type": "workshop",
      "id": 8,
      "title": "React 101",
      "userId": 1,
    },
  ],
  "message": "Events fetched successfully",
  "statusCode": 200,
  "success": true,
}
`;

exports[`Event GET /api/v1/events 1`] = `
{
  "data": [
    {
      "date": "2024-12-01",
      "description": "A discussion about the future of AI and its impact on society.",
      "event_type": "talk",
      "id": 6,
      "title": "Tech Talk: AI in 2024",
      "userId": 3,
    },
    {
      "date": "2024-12-05",
      "description": "Hands-on coding session for developers new to JavaScript.",
      "event_type": "workshop",
      "id": 7,
      "title": "JavaScript Workshop",
      "userId": 2,
    },
    {
      "date": "2024-12-10",
      "description": "A beginner's guide to building web apps with React.",
      "event_type": "workshop",
      "id": 8,
      "title": "React 101",
      "userId": 1,
    },
  ],
  "message": "Events fetched successfully",
  "statusCode": 200,
  "success": true,
}
`;
